### DISCUSSION TITLE
code-server over Jupyter (Google Colaboratory)

### DISCUSSION THREAD

**Discussion by sr229** [coder/code-server#2133] [Show and tell]
I know there's a lot of madlads out there who wants to run VS Code over Jupyter (fun fact: VS Code itself can read Jupyter Notebooks so this is Jupyterception by this point).

if I remember correctly there was a issue about running this on Jupyter so I took the liberty to grab the knowledge I got from those past issues and made it into a Google Colab Jupyter Notebook:

[![Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/sr229/vscode-jupyter/blob/master/vscode.ipynb)

## But why?

I got drunk in coffee one night and decided to read a lot of the old issues back at 2019. Mistakes were made.

## Does it really work?

Yes, but you do need to have a ngrok account otherwise VSCode can't run properly as the free tunnels with no account associations are limited to 20 connections per minute, you need 40 connections per minute at the very least.

**Comment by bpmct** [comment]
This is awesome! Thanks for sharing. Wondering if efforts can be combined with https://github.com/abhishekkrthakur/colabcode 🤔

**Reply by sr229** [reply]
Never heard of that one, definitely can be merged with that, I'll think what things can be merged in on there.

**Reply by bpmct** [reply]
also, I am working on https://github.com/bpmct/awesome-code-server so that we can put a lot of this stuff in one place. let me know if I'm missing anything. I believe you have some projects with it as well?

**Reply by sr229** [reply]
@bpmct Well kind of! I was porting code-server for Eclipse Che usage beforehand. It's still in there just lacks updates.
